src/lib/sign.c

```diff
@@ -281,20 +277,29 @@ const char *flux_sign_wrap (flux_security_t *ctx,
     }
     if (!(sign = sign_init (ctx)))
         return NULL;
+    if (!mech_type)
+        mech_type = cf_string (cf_get_in (sign->config, "default-type"));
+    if (!validate_mech (ctx, mech_type, &mech))
```

Comment:
Question:
If `!mech_type` then is `validate_mech()` redundant here? Could maybe be a crash if `default-type` not set in `sign->config` and `NULL` mech_type` is passed to `validate_mech()`?

Reply:
following on that same thought `validate_mech()` should probably just return false or assert() if name == NULL.

Reply:
Yep, `validate_mech()`should be changed to `lookup_mech()` since the mech was already validated in `sign_create()`.   Fix coming.

If `default-type` is not set, the `cf_check()` call in `sign_create()` will fail, since `.required = true` for that key in `sign_opts[]`.  So it should be safe to access it without checking later on.

Reply:
> If default-type is not set, the cf_check() call in sign_create() will fail, since .required = true for that key in sign_opts[]. So it should be safe to access it without checking later on.

Ah, ok. Makes sense. Also, as long as `validate_mech()` is never called with data from the user it should be ok to skip a check for NULL argument.

Also looks like `validate_mech` is always called on the `default-type` in `sign_create` -- I guess it doesn't hurt to do it again when a user calls `flux_sign_wrap()` using `mech_type == NULL`, but since this is going to be the common case I thought why not drop it (sorry if I wasn't clear in my initial question)?